hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -238,7 +249,11 @@ case class HoodieFileIndex(
       case (_, _) =>
         // Fetch and store latest base files and its sizes
         cachedAllInputFileSlices = partitionFiles.map(p => {
-          (p._1, fileSystemView.getLatestFileSlices(p._1.partitionPath).iterator().asScala.toSeq)
+          val fileSlices = queryInstant
```

Comment:
Just so I understand correctly. Here, we are changing existing behavior. Even if not for time travel query, previsouly we were calling fileSystemView.getLatestFileSlices. But now, since we assign queryInstant upfront (either to specified query or latest instant), we will call fileSystemView.getLatestFileSlicesBeforeOrOn. 

Is that intentional ?

Reply:
I will change the code to keep the origin behavior for non-time travel query.